script.js

```diff
@@ -0,0 +1,336 @@
+        const textareaLength = this.elements.textarea.value.length;
+        this.props.textValue = this.elements.textarea.value.substring(0, textareaLength - 1);
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'Delete') {
+        const textareaLength = this.elements.textarea.value.length;
+        this.props.textValue = this.elements.textarea.value.substring(0, textareaLength - 1);
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'Tab') {
+        this.props.textValue += '    ';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'Enter') {
+        this.props.textValue += '\n';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'Space') {
+        this.props.textValue += ' ';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'CapsLock') {
+        this.elemUpDownKeys();
+      } else if (event.code === 'ShiftLeft' || event.code === 'ShiftRight') {
+        this.elemUpDownKeys();
+      } else if (event.code === 'ArrowUp') {
+        this.props.textValue += '▲';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'ArrowLeft') {
+        this.props.textValue += '◄';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'ArrowRight') {
+        this.props.textValue += '►';
+        this.inTextArea(this.props.textValue);
+      } else if (event.code === 'ArrowDown') {
+        this.props.textValue += '▼';
+        this.inTextArea(this.props.textValue);
+      } else if (event.shiftKey && event.code === 'ControlLeft') {
+        this.changeLanguage();
+      }
+    });
+
+
+    document.body.addEventListener('keyup', (event) => {
+      event.preventDefault();
+      const key = this.elements.keysContainer.querySelector(`.keyboard__note[id = "${event.code}"]`);
+      key.classList.remove('active');
+
+      if (event.code === 'ShiftLeft' || event.code === 'ShiftRight') {
+        this.elemUpDownKeys();
+      }
+    });
+  }
+
+  inTextArea(value) {
+    this.elements.textarea.value = value || '';
+    document.querySelector('textarea').value = this.elements.textarea.value;
+  }
+
+  elemUpDownKeys() {
+    if (this.props.lang === this.keyLayout.eng) {
+      this.elements.keys.forEach((item, i) => {
+        const element = item;
+        element.textContent = this.keyLayout.engShift[i];
+      });
+      this.props.lang = this.keyLayout.engShift;
+      window.localStorage.setItem('lang', 'engShift');
+      this.elements.keys = this.elements.keysContainer.querySelectorAll('.keyboard__note');
+    } else if (this.props.lang === this.keyLayout.ru) {
+      this.elements.keys.forEach((item, i) => {
+        const element = item;
+        element.textContent = this.keyLayout.ruShift[i];
+      });
+      this.props.lang = this.keyLayout.ruShift;
+      window.localStorage.setItem('lang', 'ruShift');
```

Comment:
It's better to create some constant object with `ruShift` and other configurations. It helps to avoid mistakes in word spelling.